Source/WebCore/dom/Document.cpp

```diff
@@ -3938,6 +3938,41 @@ ExceptionOr<void> Document::write(Document* entryDocument, SegmentedString&& tex
     return { };
 }
 
+ExceptionOr<void> Document::write(Document* entryDocument, FixedVector<std::variant<RefPtr<TrustedHTML>, String>>&& strings)
+{
+    if (!isHTMLDocument() || m_throwOnDynamicMarkupInsertionCount)
```

Comment:
This exception needs moving to throw underneath the TT check to pass a test added in https://github.com/WebKit/WebKit/pull/27880 same in the writeln version

Reply:
done by moving to common shared method